src/add-edit-remove.js

```diff
@@ -0,0 +1,188 @@
+/* eslint-disable no-use-before-define */
+
+import { setLocalStorage, todosTasks, changeTodosTasks } from './myLocalStorage.js';
+
+const addDeleteIcon = (span) => {
+  const parentLi = span.parentNode;
+  const deleteSpanIcon = document.createElement('span');
+
+  deleteSpanIcon.classList.add('fas', 'fa-trash-alt', 'delete-icon');
+
+  if (parentLi) {
+    parentLi.appendChild(deleteSpanIcon);
+  }
+
+  deleteSpanIcon.addEventListener('click', () => {
+    removeTask(parentLi);
+  });
+};
+
+const removeDeleteIcon = (deleteIcon) => {
+  const parentLi = deleteIcon.parentNode;
+  if (parentLi) {
+    parentLi.removeChild(deleteIcon);
+  }
+};
+
+const changeInputToSpan = (inputDescription, hasTheLineThroughClass) => {
+  const parentLi = inputDescription.parentNode;
+
+  const span = document.createElement('span');
+  span.classList.add('description');
+
+  inputDescription.insertAdjacentElement('beforebegin', span);
+  span.innerHTML = inputDescription.value;
+
+  if (hasTheLineThroughClass) {
+    span.classList.add('line-through');
+  }
+
+  if (parentLi) {
+    parentLi.removeChild(inputDescription);
+
+    if (parentLi.classList.contains('input-description-bg')) {
+      parentLi.classList.remove('input-description-bg');
+    }
+  }
+
+  const allSpan = document.querySelectorAll('.description');
+
+  allSpan.forEach((span) => {
+    span.addEventListener('click', () => {
+      addDeleteIcon(span);
+      editTask(span);
+    });
+  });
+};
+
+const addTask = (todosTasks, value) => {
+  const newTask = {};
+  newTask.description = value;
+  newTask.completed = false;
+  newTask.index = todosTasks.length + 1;
+
+  return newTask;
+};
+
+const removeTask = (parentLi) => {
+  const taskFromLi = todosTasks.find((task) => task.index === +parentLi.id);
+
+  const newTodos = todosTasks.filter((task) => task.index !== taskFromLi.index);
+
+  const [deletedTodo] = todosTasks.filter((task) => task.index === taskFromLi.index);
+
+  newTodos.forEach((todo) => {
+    if (todo.index > deletedTodo.index) {
+      todo.index -= 1;
+    }
+  });
+
+  changeTodosTasks(newTodos);
+
+  const ul = parentLi.parentNode;
+
+  ul.removeChild(parentLi);
+
+  setLocalStorage();
+};
+
+const removeAllCompletedTasks = () => {
```

Comment:
[optional] When I remove one of the first elements, the index of one of the accessed arrays will not properly update asynchronously, and the last element will not be removed as well as the very next element will not change to complete.

Reply:
Awesome, thanks for finding this one out. It was tricky to fix since I knew it has something to do with how the index were being re-assigned after one task is deleted and after some troubleshooting I was able to get with the right solution. Feel free to try to break it again. 😆😎